src/lib/storage/chunk.cpp

```diff
@@ -1,31 +1,67 @@
-#include "chunk.hpp"
+#include <boost/preprocessor/seq/for_each.hpp>
 
 #include "abstract_segment.hpp"
+#include "all_type_variant.hpp"
+#include "chunk.hpp"
 #include "utils/assert.hpp"
+#include "value_segment.hpp"
 
 namespace opossum {
 
 void Chunk::add_segment(const std::shared_ptr<AbstractSegment> segment) {
-  // Implementation goes here
+  // FIXME: shared_ptr is copied each time, we could remove the `const` and would technically not change the
+  //        interface, but not sure if upstream is okay with that
+  _segments.emplace_back(segment);
+}
+
+template <typename T>
+static bool try_to_append_to_concrete_value_segment(AbstractSegment* segment, const AllTypeVariant& value) {
+  const auto concrete_value_segment = dynamic_cast<ValueSegment<T>*>(segment);
+  if (concrete_value_segment == nullptr) {
+    return false;
+  }
+  concrete_value_segment->append(value);
+  return true;
 }
 
 void Chunk::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
+  DebugAssert(values.size() == _segments.size(), "Tried to append row with unfitting number of columns.");
+
+  for (size_t i = 0, size = values.size(); i < size; ++i) {
```

Comment:
```suggestion
  for (auto segment_index = size_t{0}, size = values.size(); segment_index < size; ++segment_index) {
```

according to style guide